tools/text_processing/split_file_to_collection/split_file_to_collection.py

```diff
@@ -198,18 +200,16 @@ def split_by_record(args, in_file, out_dir, top, ftype):
                         newfiles[new_file_counter].write(header)
                         fresh_files.remove(new_file_counter)
                     
-                    if ftype != "sdf" and args["split_after"] == False:
-                        # write record to file
-                        newfiles[new_file_counter].write(record)
-
-                        # if not the first time through, we assign the new record
-                        record = line
-                                                
-                    else:  # for sdf we want to write the line to the record before starting a new one
+                    if not sep_at_start:
```

Comment:
isn't it preferable to make this conditional dependent on the filetype where possible? e.g. sdf should always be split after the separator, fasta always before. For generic files this is fine though.

Reply:
in my opinion an automatic method is preferable (given that it works and is simple .. which I hope is the case), because it removes a bit of code maintenance complexity

Reply:
Explicit is better than implicit! https://www.python.org/dev/peps/pep-0020/#id3

:)

Reply:
But something like `if ftype == 'sdf' or (ftype='generic' and not sep_at_start)` is hopefully not too complex.

If the splitting should always be done the same way for a particular ftype (definitely the case for fasta and sdf at least) then we can do it without needing to check the first line.

Reply:
to be clear I agree we can remove the split_after param, I just think the automatic method should take the filetype into account

Reply:
> Explicit is better than implicit!

:) true but I think this is only about how code should be written .. 

> to be clear I agree we can remove the split_after param, I just think the automatic method should take the filetype into account

but: for all filetypes that are implemented at the moment the automatic method comes to the same decision. so `if ftype == 'sdf' or (ftype == 'generic' and not sep_at_start`) is equivalent to `if not sep_at_start` .. isn't it.

> then we can do it without needing to check the first line.

reading one line should not hurt

at the moment the whole file is read anyway (twice) in any case (also changed by my my commits .. but I can undo this).

Reply:
> for all filetypes that are implemented at the moment the automatic method comes to the same decision
this is assuming all files are perfectly formatted though, if a user uploads e.g. a badly-formatted sd-file which starts with `$$$$` I would prefer the tool to do the 'correct' thing

Reply:
okidoki. will change it back.